pkg/network/node/node.go

```diff
@@ -263,6 +265,58 @@ func GetLinkDetails(ip string) (netlink.Link, *net.IPNet, error) {
 	return nil, nil, ErrorNetworkInterfaceNotFound
 }
 
+func (node *OsdnNode) validateMTU() error {
+	klog.V(2).Infof("Checking default interface MTU")
+
+	// Get the interface with the default route
+	// TODO(cdc) handle v6-only nodes
+	routes, err := netlink.RouteList(nil, netlink.FAMILY_V4)
+	if err != nil {
+		return e.New("could not list routes")
+	}
+	if len(routes) == 0 {
+		return e.New("got no routes")
+	}
+
+	const maxMTU = 65536
+	mtu := maxMTU + 1
+	for _, route := range routes {
+		// Skip non-default routes
+		if route.Dst != nil {
+			continue
+		}
+		link, err := netlink.LinkByIndex(route.LinkIndex)
+		if err != nil {
+			return fmt.Errorf("could not retrieve link id %d", route.LinkIndex)
+		}
+
+		newmtu := link.Attrs().MTU
+		if newmtu > 0 && newmtu < mtu {
+			mtu = newmtu
+		}
+	}
+	if mtu > maxMTU {
+		return e.New("unable to determine MTU")
+	}
+
+	if int(node.mtu) > mtu+50 {
+		klog.V(2).Infof("Configured node MTU is more than default interface MTU plus VXLAN overhead, tainting node...")
+		kNode, err := node.kClient.CoreV1().Nodes().Get(node.hostName, metav1.GetOptions{})
+
+		if err != nil {
+			return e.New("could not get Kubernetes Node object by hostname")
```

Comment:
`return fmt.Errorf("could not get Node object by name: %v", err)`

This prints the underlying error, along with useful commentary (since Go doesn't have real exceptions).